ecc_enclave/enclave/crypto.cpp

```diff
@@ -111,37 +109,46 @@ std::string extract_encoded_public_key(const ByteArray cert_pem)
     EVP_PKEY* pubkey = X509_get_pubkey(cert);
     EC_KEY* eckey = EVP_PKEY_get1_EC_KEY(pubkey);
 
-    BIO_ptr pkBio(BIO_new(BIO_s_mem()), BIO_free_all);
-    PEM_write_bio_EC_PUBKEY(pkBio.get(), eckey);
-    int keylen = BIO_pending(pkBio.get());
+    BIO* pkBio = BIO_new(BIO_s_mem());
+    PEM_write_bio_EC_PUBKEY(pkBio, eckey);
+    int keylen = BIO_pending(pkBio);
     ByteArray pem_str(keylen + 1);
-    int r = BIO_read(pkBio.get(), pem_str.data(), keylen);
+    int r = BIO_read(pkBio, pem_str.data(), keylen);
     if (r <= 0)
     {
         throw std::runtime_error("cannot serialize public key");
     }
     pem_str[keylen] = '\0';
+    BIO_free_all(pkBio);
 
+    EVP_PKEY_free(pubkey);
+    EC_KEY_free(eckey);
     X509_free(cert);
+    BIO_free_all(certBio);
+
     return std::string((const char*)pem_str.data(), pem_str.size());
 }
 
 std::string extract_subject_from_cert(const ByteArray cert_pem)
 {
-    BIO_ptr certBio(BIO_new(BIO_s_mem()), BIO_free_all);
-    BIO_write(certBio.get(), cert_pem.data(), cert_pem.size());
+    BIO* certBio = BIO_new(BIO_s_mem());
```

Comment:
The return value is not checked. Should doublecheck whether it fails properly later in case of error.

Reply:
added